src/wallet/rpcwallet.cpp

```diff
@@ -2678,11 +2678,12 @@ static UniValue createwallet(const JSONRPCRequest& request)
     }
     SecureString passphrase;
     passphrase.reserve(100);
+    std::string warning;
     if (!request.params[3].isNull()) {
         passphrase = request.params[3].get_str().c_str();
         if (passphrase.empty()) {
-            // Empty string is invalid
-            throw JSONRPCError(RPC_WALLET_ENCRYPTION_FAILED, "Cannot encrypt a wallet with a blank password");
+            // Empty string means unencrypted
+            warning = "Empty string given as passphrase, wallet will not be encrypted.";
```

Comment:
Couldn't happen at the moment but in the future this warning could be cleared/overwritten after `CreateWallet(...)`. Maybe the response should have `"warnings": [...]`.

Reply:
The only warnings come from wallets that already exist, so this shouldn't effect `CreateWallet`. I guess if such a warning is added that would be part of `CreateWallet` we could change it them. However it would be a breaking change to the API.

Reply:
Well it's still possible with concurrent `createwallets`. You could have 2 `std::string warn1, warn2;` and at the end join them :hear_no_evil:

Reply:
I agree with promag that we shouldn't just throw away the first `warning` string. If you're worried about breaking the API, then joining the strings seems like good solution.

Reply:
Joined them